lib/wallet/wallet.js

```diff
@@ -1736,6 +1740,52 @@ class Wallet extends EventEmitter {
     return mtx.signAsync(rings, Script.hashType.ALL, this.wdb.workers);
   }
 
+  /**
+   * Count all pending ancestors.
+   * @param {TX} tx
+   * @returns {Promise} - Returns {Number}
+   */
+
+   async countPendingAncestors(tx) {
```

Comment:
I'm wondering if knowing the set of pending ancestors would be useful at all decoupled from its size, ie making this function return a `BufferSet` instead of a `Number`. The consuming code could just call `set.size` on the returned `Set` and have the same functionality, and other code that wants to introspect on the set of ancestors could do so more easily without needing to add an additional function.

Reply:
Ok, agreed, updated at: https://github.com/bcoin-org/bcoin/pull/873/commits/bb55b14eb3036a1e5bebbeba236564268dc5f1de

So the API is `await wallet.getPendingAncestors(tx)` which will return a set of TX hashes that the queried TX depends on.